zio/zjsonio/reader.go

```diff
@@ -165,6 +165,34 @@ func decodeType(columns []interface{}) (string, error) {
 	return s + "]", nil
 }
 
+func decodeField(builder *zcode.Builder, typ zng.Type, s string) error {
+	var err error
+	var index int
+	b := []byte(s)
+	if zng.IsContainerType(typ) {
+		return zng.ErrNotContainer
+	}
+	if utyp, ok := typ.(*zng.TypeUnion); ok {
+		b, typ, index, err = utyp.SplitZng(b)
+		if err != nil {
+			return err
+		}
+		builder.BeginContainer()
+		defer builder.EndContainer()
+		var a [8]byte
+		n := zcode.EncodeCountedVarint(a[:], int64(index))
```

Comment:
Should this be zcode.EncodeCountedUvarint?

Reply:
That seems like a good idea, changing.